DscResources/MSFT_xWindowsUpdate/MSFT_xWindowsUpdate.psm1

```diff
@@ -137,7 +137,13 @@ function Set-TargetResource
     }
     $uri, $kbId = Validate-StandardArguments -Path $Path -Id $Id
 
-    
+    #check if there is currently an update pending, if so error
+    Write-Verbose "Checking wusa for pending reboots"
+    $wua = New-Object -ComObject Microsoft.Update.SystemInfo
+    if($wua.RebootRequired)
+    {
+        Throw "There is a prior reboot pending. This prevents wusa from being able to install $Id."
```

Comment:
I think we should add this `$global:DSCMachineStatus = 1`  here as well, in side the brackets, before the throw.